### DISCUSSION TITLE
Questionnaires get displayed twice

### DISCUSSION THREAD

**Discussion by katjahaider** [KL-Psychological-Methodology/ESMira#5] [Questions] [answered]
Hello! We have a single person in the study who has had all questionnaires displayed twice since the last update of the study. The problem does not exist for all other participants. I have seen in the events that "study updated" is included twice at exactly the same time (date, hour, minute, and second) and the same applies to all "invitations" for questionnaires. What could be the reason for this? Is there a way to fix the error for this one person? Clicking on "Update studies" again did not work. Thank you for your support!

**Comment by JodliDev** [comment]
My apologies for my late response. My email notifications seem to have betrayed me.

Hm. I never encountered a problem like this (and I am unsure what could have caused this).
Would it be possible to have the participant send an error report (via the gear icon at the top right).
Also a screenshot might help.

What happens if one of the questionnaires are filled out (or both)? Is data still sent to the server correctly?

**Comment by katjahaider** [comment]
No worries! Thank you for your reply.

The participant has sent the following screenshot. Both "Handynutzung" and "Umfrage lang" were displayed twice originally. "Umfrage lang" has already been completed here once, which is why it is only visible once on the screenshot. 

Below I have also attached a screenshot from the events file so that you can see that the participant always receives the invitations twice since the update to study version 44.0. 

I have also asked the participant to send an error report and assume that they will do so in the next few days. 

So far, I have agreed with the participants that they should only complete the surveys once despite being invited twice, and so far we have received a correct set of data. So it's not a problem in terms of the data, it's just inconvenient for the participant that they are always notified twice and that the surveys are also displayed twice in the app. 

Thank you in advance for your help! 

![admin](https://github.com/KL-Psychological-Methodology/ESMira/assets/151832553/0bd36ede-1e50-4c13-a3a3-b2a65eeec8dd)
![events screenshot](https://github.com/KL-Psychological-Methodology/ESMira/assets/151832553/18a061b9-d362-4e01-be90-d6e46fd3d2ea)

**Comment by JodliDev** [comment]
Hm. And you are saying that others received the same study update and it did not cause the same issue?
Do you remember what kind of study update it was? Like, did the order of questionnaires change? Or were new questionnaires added or removed? Or were visibility settings (e.g. when it is active) of questionnaires changed?

So, usually, when updating a study, the app compares the old and new version, and if there were too many changes (or changes that are hard to track because they can have multiple side effects), the app decides to just delete the old study and recreate it.
So my theory is, that for some reason, the "old study" was not fully deleted. But I have no idea how that can happen. I am hoping I can see more in the error report.

Right now there is no convenient way of forcing an app do redownload an already downloaded study version. But you can just change something very simple (e.g. a whitespace in the study description) in the study, save it and update the study version.
This will force their app to download the new version (and hopefully delete the additional questionnaire).
Everyone else will also download the new version. But since almost nothing changed, it should change anything for them.

**Comment by katjahaider** [comment]
Yes, the others received the same study update and it didn't cause any problems. 
In the latest update, we added and removed some questions and also changed something in the visibility settings. 

I want to be careful with changes and new updates at the moment because the settings are just right and I don't want to have to encourage the participants to update too often. But the schedules need to be adjusted again at the beginning of the year anyway. Can I assume that this new update will also solve the problem for this one participant?

**Comment by JodliDev** [comment, marked as answer]
So I had a look at the error report. My theory is:
As far as I can tell, the faulty update happened on 17.11.23.
It looks to me, like the update process has been triggered twice at the same time (which should not be possible but is not prevented properly). And since it was a major update and the study had to be recreated, this lead to the study content being recreated twice.
I added a fix for the app (which should be online for participants in the next couple weeks) that should prevent such an occurrence in the future.

If my theory is correct, then it needs a study update, that triggers a recreating of the study to fix the problem for the participant. So changing the schedules should resolve the problem for them, yes.

Btw
You dont have to ask your participants to update manually (in fact I assume that this error was caused by them pressing the update button several times too quickly in succession, which the app did not prevent properly). The "update studies" button mainly exists for testing purposes. Participants should never have to use it.
When starting ESMira, the app automatically searches for new updates (and a background service also checks for updates twice a day)
